core/lib/utils.ts

```diff
@@ -1,2 +1,20 @@
 export const noop = () => {};
 export const identity = <T>(x: T) => x;
+
+/**
+ * @param number - decimal number
+ * @returns the decimal precision of the number
+ */
+export function getDecimalPrecision(number: number) {
+	const regexp = ('' + number).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
+	if (!regexp) {
+		return 0;
+	}
+	return Math.max(
+		0,
+		// Number of digits right of decimal point.
```

Comment:
I think the code can be simplified with: `(regexp[1]?.length ?? 0)`

Reply:
and the second one can be (+matches[2] || 0) I believe

Reply:
No, sorry, I re-changed my original code as I saw ts errors locally. The + must be outside the parenthesis as +undefined gives NaN, and the ?? needs undefined to work well

Reply:
as discussed +Nan || 0 will give 0, so should work fine